jobs/batch_operation_sample.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START instantiate]
+from googleapiclient.discovery import build
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START batch_job_create]
+def batch_job_create(client_service, company_name):
+    import base_job_sample
+    created_jobs = []
+
+    def job_create_callback(request_id, response, exception):
+        if exception is not None:
+            print('Got exception while creating job: %s' % exception)
+            pass
+        else:
+            print('==========\nJob created: %s\n==========' % response)
+            created_jobs.append(response)
+            pass
+
+    batch = client_service.new_batch_http_request()
+    job_to_be_created1 = base_job_sample.generate_job_with_required_fields(
+        company_name)
+    request1 = {'job': job_to_be_created1}
+    batch.add(
+        client_service.jobs().create(body=request1),
+        callback=job_create_callback)
+
+    job_to_be_created2 = base_job_sample.generate_job_with_required_fields(
+        company_name)
+    request2 = {'job': job_to_be_created2}
+    batch.add(
+        client_service.jobs().create(body=request2),
+        callback=job_create_callback)
+    batch.execute()
+
+    return created_jobs
+
+
+# [END batch_job_create]
+
+
+# [START batch_job_update]
+def batch_job_update(client_service, jobs_to_be_updated):
+    updated_jobs = []
+
+    def job_update_callback(request_id, response, exception):
+        if exception is not None:
+            print('Got exception while updating job: %s' % exception)
+            pass
+        else:
+            print('==========\nJob updated: %s\n==========' % response)
+            updated_jobs.append(response)
+            pass
+
+    batch = client_service.new_batch_http_request()
+    for index in range(0, len(jobs_to_be_updated)):
```

Comment:
Side note: if you switch to updating all jobs instead, this part can become `for job in jobs`, which is easier to understand and more idiomatic for Python devs.

Reply:
I know. However, as mentioned above, they want half jobs updated using field mask feature, and half not.